### PR TITLE
Addon implementations for TransactionListFunctions + Logger

### PR DESCRIPTION
**Pull request by FeathersRe**
Edited TransactionListFunctions to fit in new definitions
Fixed JUnit Test: TransactionListTest such that it successfully checks program output
Added LongAhlogger to keep track of program process in LongAh main()

### CODE REVIEW

File: log/LongAh.log [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Mar 20, 2024 3:50:24 AM longah.LongAh main
```
Comment (1simjustin) [inline log/LongAh.log:1]: can you go to `.gitignore` and add `log/` at the bottom

**Review by 1simjustin — COMMENTED** [review]
looks good! just help me make changes to the gitignore so we don't flood this repo with logs

File: log/LongAh.log (hunk 1 above)
Comment (FeathersRe) [reply log/LongAh.log]: Edited!

**Review by 1simjustin — APPROVED** [review]
lgtm, thanks!

**Pull request merged**